### DISCUSSION TITLE
Multiple portals in the DOM

### DISCUSSION THREAD

**Discussion by Sergey7709** [KevinVandy/mantine-react-table#264] [Q&A] [answered]
Please clarify why the table creates so many <div data-portal="true"></div> in the DOM? Is there any way to change this?
![4](https://github.com/KevinVandy/mantine-react-table/assets/116739566/59253a4f-2c19-4766-b6a8-22beaf650466)

**Comment by KevinVandy** [comment, marked as answer]
They're for tooltips, popovers, menus and modals. You can lower the amount of these portals by making all Mantine tooltips share 1 portal. In my Mantine theme, I do:

```ts
<MantineProvider
  theme={createTheme({
    //...
    components: {
      //...
      Tooltip: {
        defaultProps: {
          withArrow: true,
          portalProps: {
            target: '.mantine-tooltips', // all tooltips share 1 portal
          },
        },
      },
      //..
    },
  })}
>
  {children}
</MantineProvider>
```

and then just make an empty div in the body of my HTML with that class.

Not really practical to do the same for Modals, Popovers and menus, as they can sometimes need to both be on screen at once, but this makes sense for Tooltips.

**Reply by Sergey7709** [reply]
Thank you!
